### PR TITLE
Enhance the capacity report to show worker registration info

### PR DESCRIPTION
**Pull request by elega**
### What changes are proposed in this pull request?

Screen Shot 2023-02-21 at 10 02 05 PM

Please outline the changes and how this PR fixes the issue.

### Why are the changes needed?

Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, describe the bug.

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
  1. change in user-facing APIs
  2. addition or removal of property keys
  3. webui

### CODE REVIEW

File: shell/src/main/java/alluxio/cli/fsadmin/report/CapacityCommand.java [hunk 1]
```diff
@@ -94,7 +101,20 @@ public int run(CommandLine cl) throws IOException {
    */
   public void generateCapacityReport(GetWorkerReportOptions options, AlluxioConfiguration conf)
       throws IOException {
-    List<WorkerInfo> workerInfoList = mBlockMasterClient.getWorkerReport(options);
+    boolean workerRegisterToAllMasters =
```
Comment (jiacheliu3) [inline shell/src/main/java/alluxio/cli/fsadmin/report/CapacityCommand.java:104]: Interestingly this command may not execute on a node which has this worker config property. But that'd be a super rare case and not worth handling specially.

File: shell/src/main/java/alluxio/cli/fsadmin/report/CapacityCommand.java [hunk 2]
```diff
@@ -94,7 +101,20 @@ public int run(CommandLine cl) throws IOException {
...
+        conf.getBoolean(PropertyKey.WORKER_REGISTER_TO_ALL_MASTERS);
+
+    final List<WorkerInfo> workerInfoList;
+    final alluxio.client.block.AllMastersWorkerInfo allMastersWorkerInfo;
+    if (workerRegisterToAllMasters) {
+      allMastersWorkerInfo =
+          alluxio.client.block.util.WorkerInfoUtil.getWorkerReportsFromAllMasters(
```
Comment (jiacheliu3) [inline shell/src/main/java/alluxio/cli/fsadmin/report/CapacityCommand.java:111]: import normally?

File: shell/src/main/java/alluxio/cli/fsadmin/report/CapacityCommand.java [hunk 3]
```diff
@@ -184,6 +207,88 @@ private void printAggregatedInfo(GetWorkerReportOptions options) {
...
+        sb.append(",");
+      }
+    }
+    sb.append("]");
+    return sb.toString();
+  }
+
+  private String masterAddressToString(
+      java.net.InetSocketAddress inetSocketAddress,
+      java.util.Collection<java.net.InetSocketAddress> masterAddresses) {
```
Comment (jiacheliu3) [inline shell/src/main/java/alluxio/cli/fsadmin/report/CapacityCommand.java:228]: import normally?

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Added a batch of comments, PTAL thanks!

File: shell/src/main/java/alluxio/cli/fsadmin/report/CapacityCommand.java (hunk 1 above)
Comment (elega) [reply shell/src/main/java/alluxio/cli/fsadmin/report/CapacityCommand.java:104]: this config has ENFORCE consistency level so this should not happen

File: core/client/fs/src/main/java/alluxio/client/block/util/WorkerInfoUtil.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        });
+        workerIdAddressMap.compute(info.getId(), (k, v) -> {
+          InetSocketAddress workerAddress =
+              InetSocketAddress.createUnresolved(info.getAddress().getHost(),
+                  info.getAddress().getRpcPort());
+          if (v == null) {
+            return workerAddress;
+          }
+          if (!v.equals(workerAddress)) {
+            LOG.warn("The same worker id {} corresponds to multiple worker name {} {}, "
```
[the remark this answers is not in the document]
Comment (elega) [reply core/client/fs/src/main/java/alluxio/client/block/util/WorkerInfoUtil.java:133]: i change the log level to ERROR. What else do you expect me to do?

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks for the work!

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks for the work!

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**